rmw_connext_cpp/src/functions.cpp

```diff
@@ -230,7 +230,7 @@ rmw_create_publisher(
     }
   }
 
-  if (!get_datawriter_qos(participant, qos_profile, datawriter_qos)) {
+  if (!get_datawriter_qos(participant, *qos_profile, datawriter_qos)) {
```

Comment:
2 things:

I think it's fine if the shared API keeps a reference to the qos_profile, because the shared API is C++. That way we can avoid duplicating code to check that the pointer is not null before dereferencing. On that note, do you need to check if qos_profile is null here before dereferencing it?